Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Get distro information
+GetDistro
+
+dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+supported_distro_list=(redhat centos suse debian ubuntu)
+
+readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+
+function test_case_cleanup(){
+    /opt/microsoft/dependency-agent/uninstall
+
+    for dir in "${dir_list[@]}"
+    do
+        rm -rf $dir
+    done
+
+    kill $watch_pid
+
+    LogMsg "Finished cleaning up test cases"
+}
+
+# Calls the cleanup function when interupted or exited
+trap test_case_cleanup SIGINT EXIT
+
+function verify_directory() {
+    failed_assertions=0
+    for dir in "${dir_list[@]}"
+    do
+        [ -d $dir ] && LogErr "Directory $dir exists." && SetTestStateFailed && failed_assertions=$[[failed_assertions+1]]
+    done
+
+    if ["$failed_assertions" -gt 0]; then
+        LogErr "Some/All directories exists"
+        SetTestStateFailed
+        exit 0
+    fi
+}
+
+function verify_distro() {
+    LogMsg "Current distro: $DISTRO"
+    for distro in "${supported_distro_list[@]}"
+    do
+        if [[ "$distro"* == $DISTRO ]]; then
+        LogMsg "Supported Distro family"
+        break
+    else
+        LogErr "Unsupported Distro family"
```

Comment:
If `$DISTRO` doesn't match the first value in `supported_distro_list` then control reaches the else block and the test is skipped.

Suggested alternatives:

- declare `supported_distro_list` as an associative array `declare supported_distro_list[suse]=true supported_distro_list[centos]=true ...`
To test `$DISTRO` use `if ${supported_distro_list[$DISTRO]:-false} ; then` _support_ `else` _unsupport_` fi` do loop needed.

- Use a case statement
```
    supported=true  # assume success
    case $DISTRO in
        redhat) ;;
        suse) ;;
        ...
        *) supported=false ;;
    esac

    if $supported ; then ... else ... fi
```

Reply:
Adopted the case statement approach